### PR TITLE
Remove permission check for Search Nav Item (3.3)

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

**This is a backport of #8917 for `3.3`.**

Prior to this change, the Search Menu Item was only displayed for admins or users which have searches:absolute', 'searches:relative', 'searches:keyword' permissions.

The mentioned permission could only be add via API requests. Which left the admin users as the only ones which could really see that Menu Item. 
 
But a user could go to that page either via a link (on the streams page) or by typing the url manually.

This change will now simply remove this check and always display the menu item. The search still will only work if the user has read permissions to a stream.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Review by bernd — COMMENTED** [review]
Does it make sense to backport this without also having https://github.com/Graylog2/graylog2-server/pull/8958 backported too? If we just backport this one, users will get the error message instead of the empty stream page if they don't have access to any streams.

**Comment by bernd** [comment]
> #8958 will be backported too, once it has been merged to `master`.

Cool, thanks. Then I will wait until that's done to merge both at the same time. That way we avoid merging only one and release before merging the other one. :smile:

File: graylog2-web-interface/src/components/navigation/Navigation.jsx
Comment (kyleknighted) [inline graylog2-web-interface/src/components/navigation/Navigation.jsx:31]: Why add the `eslint-disable` instead of destructing `URLUtils`?

Comment (dennisoelkers) [reply graylog2-web-interface/src/components/navigation/Navigation.jsx:31]: To keep the change for the backport as small as possible.

**Comment by dennisoelkers** [comment]
@bernd: #8958 has been merged and is now backported as #9011.

**Comment by bernd** [comment]
> @bernd: #8958 has been merged and is now backported as #9011.

@dennisoelkers Awesome, thank you!

**Review by bernd — APPROVED** [review]

**Pull request merged**
